syntax = "proto3";
package com.grpc.client;

option java_multiple_files = true;
option java_package = "com.grpc.client";
option java_outer_classname = "ModelServiceProto";
//option objc_class_prefix = "HLW";

service ModelService {
  // rpc PredictOptimalParameters(ParameterRequest) returns (ParameterResponse) {}
  rpc UpdatePerformanceData(PerformanceData) returns (UpdateResponse) {}
}

// message ParameterRequest {
//   int32 current_merge_size = 1;
//   int32 current_thread_num = 2;
//   double current_throughput = 3;
// }

// message ParameterResponse {
//   int32 optimal_merge_size = 1;
//   int32 optimal_thread_num = 2;
//   double expected_improvement = 3;
// }

message PerformanceData {
  repeated DataPoint historical_data = 1;
}

message DataPoint {
  int32 merge_size = 1;
  int32 thread_num = 2;
  double throughput = 3;
}

message UpdateResponse {
  bool success = 1;
  Prediction prediction = 2;
}

message Prediction {
  int32 optimal_merge_size = 1;
  int32 optimal_thread_num = 2;
  double expected_throughput = 3;
}